extendr-api/src/metadata.rs

```diff
@@ -215,25 +230,16 @@ fn write_method_wrapper(
         return Ok(());
     }
 
-    let actual_args = func
-        .args
-        .iter()
-        .map(|arg| sanitize_identifier(arg.name))
-        .collect::<Vec<_>>();
-    let formal_args = if !actual_args.is_empty() && actual_args[0] == "self" {
-        // Skip a leading "self" argument.
-        // This is supplied by the environment.
-        actual_args
-            .iter()
-            .skip(1)
-            .map(|x| x.to_string())
-            .collect::<Vec<_>>()
-    } else {
-        actual_args.clone()
-    };
+    let r_args: Vec<RArg> = func.args.iter().map(Into::into).collect();
+    let actual_args = r_args.iter().map(|a| a.to_actual_arg());
+
+    // Skip a leading "self" argument.
+    // This is supplied by the environment.
+    let n_skip = r_args
+        .first()
+        .map_or(0, |a| if a.is_self() { 1 } else { 0 });
 
-    let formal_args = formal_args.join(", ");
-    let actual_args = actual_args.join(", ");
+    let formal_args = r_args.iter().skip(n_skip).map(|a| a.to_formal_arg());
```

Comment:
Could we not express this more concisely by just `.filter()`ing this instead of using `.skip()`?

Reply:
I propose a middle ground in form of `.skip_while(|a| a.is_self())`. This communicates the idea that we want to drop the `self` argument found at the beginning of the argument list.

Reply:
Sounds good to me.